File: backend/app/schemas/task.py
from pydantic import BaseModel, Field


class ScheduleConfig(BaseModel):
    cron: str | None = None
    interval_minutes: int | None = None


class ExecutionContent(BaseModel):
    meter_ids: list[int] | None = None
    obis_codes: list[str] | None = None
    analysis_type: str | None = None
    report_type: str | None = None


class TaskCreate(BaseModel):
    task_name: str
    task_type: str
    schedule_config: ScheduleConfig = Field(default_factory=ScheduleConfig)
    execution_content: ExecutionContent = Field(default_factory=ExecutionContent)
    filter_config: dict = Field(default_factory=dict)
    priority: int = 3
    retry_times: int = 3
    timeout: int = 60
    is_enabled: bool = True


class TaskUpdate(BaseModel):
    task_name: str | None = None
    task_type: str | None = None
    schedule_config: dict | None = None
    execution_content: dict | None = None
    filter_config: dict | None = None
    priority: int | None = None
    retry_times: int | None = None
    timeout: int | None = None
    is_enabled: bool | None = None


class TaskToggle(BaseModel):
    is_enabled: bool

File: frontend/apps/web-antd/src/api/modules/task.ts
import { requestClient } from '#/api/request';

export interface TaskListParams {
  page?: number;
  page_size?: number;
  status?: string;
  task_type?: string;
  task_category?: string;
}

export interface TaskFormData {
  task_name: string;
  task_category: 'collection' | 'analysis' | 'report' | 'cleanup';
  task_type: 'cron' | 'interval' | 'once';
  schedule_config: Record<string, any>;
  execution_content: Record<string, any>;
  filter_config: Record<string, any>;
  priority?: number;
  retry_times?: number;
  timeout?: number;
  is_enabled?: boolean;
}

export function getTaskList(params: TaskListParams) {
  return requestClient.get('/tasks', { params });
}

export function getTaskDetail(id: number) {
  return requestClient.get(`/tasks/${id}`);
}

export function createTask(data: TaskFormData) {
  return requestClient.post('/tasks', data);
}

export function updateTask(id: number, data: Partial<TaskFormData>) {
  return requestClient.put(`/tasks/${id}`, data);
}

export function deleteTask(id: number) {
  return requestClient.delete(`/tasks/${id}`);
}

export function executeTask(id: number, data?: { device_ids?: number[] }) {
  return requestClient.post(`/tasks/${id}/execute`, data || {});
}

export function toggleTask(id: number, enabled: boolean) {
  return requestClient.request(`/tasks/${id}/toggle`, { data: { is_enabled: enabled }, method: 'PATCH' });
}

export function getTaskLogs(params: { task_id: number; page?: number; page_size?: number }) {
  return requestClient.get(`/tasks/${params.task_id}/logs`, {
    params: { page: params.page, page_size: params.page_size },
  });
}

export function getTaskDeviceLog(logId: number, params?: { status?: string }) {
  return requestClient.get(`/task-logs/${logId}/devices`, { params });
}
